aws-network-acl-allows-public-ingress.tf

```
@@ -0,0 +1,1439 @@
+  }
+}
+
+# ruleid: aws-network-acl-allows-public-ingress
+resource "aws_network_acl_rule" "fail_end_outside_link_local_1" {
+  egress      = false
+  protocol    = "tcp"
+  from_port   = 22
+  to_port     = 22
+  rule_action = "allow"
+  cidr_block  = "169.254.0.0/15" # 169.254.0.0 - 169.255.255.255
+}
+
+resource "aws_network_acl" "fail_end_outside_link_local_1" {
+  vpc_id = aws_vpc.example.id
+
+  # ruleid: aws-network-acl-allows-public-ingress
+  ingress {
+    rule_no    = 1
+    protocol   = "tcp"
+    from_port  = 22
+    to_port    = 22
+    action     = "allow"
+    cidr_block = "169.254.0.0/15" # 169.254.0.0 - 169.255.255.255
+  }
+}
+
+# ruleid: aws-network-acl-allows-public-ingress
+resource "aws_network_acl_rule" "fail_end_outside_link_local_2" {
+  egress      = false
+  protocol    = "tcp"
+  from_port   = 22
+  to_port     = 22
+  rule_action = "allow"
+  cidr_block  = "169.254.99.99/15" # 169.254.0.0 - 169.255.255.255
+}
+
+resource "aws_network_acl" "fail_end_outside_link_local_2" {
+  vpc_id = aws_vpc.example.id
+
+  # ruleid: aws-network-acl-allows-public-ingress
+  ingress {
+    rule_no    = 1
+    protocol   = "tcp"
+    from_port  = 22
+    to_port    = 22
+    action     = "allow"
+    cidr_block = "169.254.99.99/15" # 169.254.0.0 - 169.255.255.255
+  }
+}
+
+# ruleid: aws-network-acl-allows-public-ingress
+resource "aws_network_acl_rule" "fail_end_outside_link_local_3" {
+  egress      = false
+  protocol    = "tcp"
+  from_port   = 22
+  to_port     = 22
+  rule_action = "allow"
+  cidr_block  = "169.254.255.255/15" # 169.254.0.0 - 169.255.255.255
+}
+
+resource "aws_network_acl" "fail_end_outside_link_local_3" {
+  vpc_id = aws_vpc.example.id
+
+  # ruleid: aws-network-acl-allows-public-ingress
+  ingress {
+    rule_no    = 1
+    protocol   = "tcp"
+    from_port  = 22
+    to_port    = 22
+    action     = "allow"
+    cidr_block = "169.254.255.255/15" # 169.254.0.0 - 169.255.255.255
+  }
+}
+
+# ruleid: aws-network-acl-allows-public-ingress
+resource "aws_network_acl_rule" "fail_end_outside_link_local_5" {
+  egress      = false
+  protocol    = "tcp"
+  from_port   = 22
+  to_port     = 22
+  rule_action = "allow"
+  cidr_block  = "169.255.0.0/15" # 169.254.0.0 - 169.255.255.255
+}
+
+resource "aws_network_acl" "fail_end_outside_link_local_5" {
+  vpc_id = aws_vpc.example.id
+
+  # ruleid: aws-network-acl-allows-public-ingress
+  ingress {
+    rule_no    = 1
+    protocol   = "tcp"
+    from_port  = 22
+    to_port    = 22
+    action     = "allow"
+    cidr_block = "169.255.0.0/15" # 169.254.0.0 - 169.255.255.255
+  }
+}
+
+# ruleid: aws-network-acl-allows-public-ingress
+resource "aws_network_acl_rule" "fail_start_outside_shared_1" {
+  egress      = false
+  protocol    = "tcp"
+  from_port   = 22
+  to_port     = 22
+  rule_action = "allow"
+  cidr_block  = "100.64.0.0/9" # 100.0.0.0 - 100.127.255.255
+}
+
+resource "aws_network_acl" "fail_end_outside_shared_1" {
```

Small mistake with the test case name:
```suggestion
resource "aws_network_acl" "fail_start_outside_shared_1" {
```